src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -26,6 +26,77 @@ public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary
 
     }
 
+    public async Task<IEnumerable<EmissionsForecast>> GetCurrentForecastDataAsync(IDictionary props)
+    {
+        DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, DateTimeOffset.Now);
+        DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End,  start.AddDays(1));
+        _logger.LogInformation("Aggregator getting carbon intensity forecast from data source");
+
+        var forecasts = new List<EmissionsForecast>();
+        foreach(var location in GetLocationOrThrow(props))
+        {
+            var forecast = await this._dataSource.GetCurrentCarbonIntensityForecastAsync(location);
+            forecast.NotBefore = start;
+            forecast.NotAfter = end;
+            forecast.ForecastData = FilterByDate(forecast.ForecastData, start, end);
+            TransformGranularity(forecast, props);
+            forecast.Optimal = GetOptimalEmissions(forecast.ForecastData);
+            forecasts.Add(forecast);
+        }
+
+        return forecasts;
+    }
+
+    private IEnumerable<EmissionsData> FilterByDate(IEnumerable<EmissionsData> data, DateTimeOffset start, DateTimeOffset end)
+    {
+        return data.Where(ed => ed.TimeBetween(start.DateTime, end.DateTime));
+    }
+
+    private void TransformGranularity(EmissionsForecast forecast, IDictionary props)
+    {
+
+        var data = forecast.ForecastData.ToList();
+        var current = data.First();
+        var location = current.Location;
+        var next = data.Skip(1).First();
+        
+        var defaultWindowSize = next.Time - current.Time;
+        if(props.Contains(CarbonAwareConstants.Duration) && props[CarbonAwareConstants.Duration] is int duration)
```

Comment:
never seen this syntax before, might want to explain what "is int duration" does. from online seems like "is" is a type checker, confused on what duration is though